pages/articles.jsx

```diff
@@ -142,64 +67,115 @@ export default function ArticlesPage({ allContentTypes }) {
+			})
+		)
+	}
+
+	// Get the content from the md file
+	async function getMarkdownFileContent(data, keepEncoded, signal) {
+		return await Promise.all(
+			data.map(async item => {
+				return await await getRepoData(`/${item.path}`, keepEncoded, signal)
+			})
+		)
+	}
+
+	async function _fetchMarkdownFiles(signal) {
+		// Returns array of all decoded config data for ALL related CT's
+		const _relatedCTConfigDataArray = allContentTypes.filter(function(i){
+			if (router.query.group) {
+				return router.query.group === i.group
+			} else {
+				return router.query.type === i.label
+			}
+		// eg: {label: "Posts", id: "posts", path: "content/posts", ...}
+		})
+		// Returns a list of only the content directories ('path') from the config array
+		const _articlesLocationsList = new CT(_relatedCTConfigDataArray).getPaths()
+
+		// fetch all the files from the directories mentioned in `_articlesLocationsList`
+		listMarkdownFiles(_articlesLocationsList, false, signal)
+			.then(markdownFilesList => {
+				// flattern the output with the list of md files from the directory
+				const markdownFilesListFormatted = markdownFilesList.flat()
+
+				// Get the content from the md file
+				getMarkdownFileContent(markdownFilesListFormatted, false, signal)
+					.then(markdownFilesContent => {
+						// format md file content into the object
+						const formattedMarkdownFilesContent = markdownFilesContent.map(content =>
+							content && matter(content)
+						)
+
+						// ADD `path` property to the markdown content `formattedMarkdownFilesContent`
+						formattedMarkdownFilesContent.forEach((item, index) => {
+							item.path = markdownFilesListFormatted[index].path
+						})
+
+						// ADD `name` property inside the `data` object
+						formattedMarkdownFilesContent.forEach((item, index) => {
+							// Delete the extension
+							let fileName = markdownFilesListFormatted[index].name.split(`.`)
+							// Delete the last item in an array (extension)
+							fileName.pop()
+							// Concatenate all the rest items in an array
+							fileName = fileName.join()
+							// Uppercase the name
+							fileName = string.titleCase(fileName)
+							// Create a `name` property at the `data`
+							item.data.name = fileName
+						})
+						setMarkdownContent(formattedMarkdownFilesContent)
+					})
+					.catch(error => {
+						console.log(error)
+					})
+			})
+			.catch(error => {
+				console.log(error)
+			})
+	}
 
 	return (
 		<>
 			<Header
-				title={title}
+				title={router.query.group || router.query.type}
 				subtitle="All related articles are listed on this page"
 			/>
 
 			{/* This creates an 'Add New' button/dropdown. It would show a dropdown of all CTs
 			in the current group and take the user to a page for adding a new article (using editor.jsx page).
 			If there's only one content-type, it wouldn't be a dropdown, just a button. */}
-			<ArticleCreate data={demo} />
+			<ArticleCreate data={markdownContent} />
```

Comment:
@IvaTsu I think we need to be passing in a list of CT's from `_relatedCTConfigDataArray` instead of the markdown content. See my comment on the `ArticlesCreate` component.